authors/apps/articles/views.py

```diff
@@ -43,6 +43,25 @@ class ArticleView(RetrieveUpdateDestroyAPIView):
     """
     permission_classes = (IsAuthenticatedOrReadOnly,)
 
+    def get(self, request, slug):
+        """
+        Retrieve a specific article
+        :param slug
+        any user should view details of an article
+        """
+        # Get the article user searched
+        # When article is not found, Give the user a description
+        # of what happened. If the article is found return it back to user
+
+        if not Article.objects.filter(slug=slug):
```

Comment:
Have the article query in a try-catch clause to catch the not found error. This reduces the number of queries to the database.

Reply:
Okay @bettblake08 let me switch to that.

Reply:
Switched out to try except, Thanks for the feedback. Please check on the update.